lib/std/hash_map.zig

```diff
@@ -508,7 +508,7 @@ pub fn HashMap(
         /// If a new entry needs to be stored, this function asserts there
         /// is enough capacity to store it.
         pub fn getOrPutAssumeCapacityAdapted(self: *Self, key: anytype, ctx: anytype) GetOrPutResult {
-            return self.unmanaged.getOrPutAssumeCapacityAdapted(self.allocator, key, ctx);
```

Comment:
This slipped presumably because we didn't touch this bit in any test, correct? Perhaps it'd be good to add a smoke test to avoid this kind of regression in the future?